cds_ils/importer/providers/cds/rules/journal.py

```diff
@@ -134,14 +140,45 @@ def migration(self, key, value):
                 "subscription": clean_val("a", value, str),
             }
         )
-    if key == "780__":
-        _relation_previous = clean_val("w", value, str, req=True)
 
-    if key == "785__":
-        _relation_next = clean_val("w", value, str, req=True)
+    try:
+        _related = _migration.get("related", [])
+        relation_type = None
+        _has_related = _migration.get("has_related", False)
+        # language relation
+        if key == "787__" and "i" in value:
+            relation_language = clean_val("i", value, str)
+            if relation_language:
+                relation_type = LANGUAGE_RELATION.name
+        # has supplement/supplement to relation
+        if key == "770__" or key == "772__" and "i" in value:
+            clean_val("i", value, str)
+            relation_type = OTHER_RELATION.name
+        # continues/is continued by relation
+        if key == "780__" or key == "785__" and "i" in value:
+            relation_sequence = clean_val("i", value, str)
+            if relation_sequence:
+                relation_type = SEQUENCE_RELATION.name
+                if key == "780__":
+                    _relation_previous = clean_val("w", value, str, req=True)
+                else:
+                    _relation_next = clean_val("w", value, str, req=True)
+
+        if relation_type:
+            _related.append(
+                {
+                    "related_recid": clean_val("w", value, str, req=True),
+                    "relation_type": relation_type,
+                }
+            )
+    except ManualImportRequired as e:
+        e.subfield = "i"
+        raise e
 
     _migration.update(
         {
+            "related": _related,
+            "has_related": _has_related or bool(relation_type),
```

Comment:
from the code above `_has_related` will be always `False`

Reply:
if there was already a relation that was migrated the `_has_related` would have been set to `True`

Reply:
in which of the previous rules it might happen?